main.py

```diff
@@ -101,9 +105,205 @@ def workflow_1(playlist_length):
+            option_label_fn=get_song_option_label_with_indent,
+            message='Select the index of the song that you wish to add: ')
+
+        if not valid_index:
+            print(f'Sorry. There is no song at index "{index}".')
+            selecting_song = input_bool("Try another index?")
+
+        else:
+            song_number = song_numbers[index - 1]
+            selecting_song = False
+
+    return song_number
+
+
+def display_seeds(seeds):
+    print(f'Seeds:')
+    for i, song_number in enumerate(seeds, start=1):
+        print(get_song_option_label(i, song_number))
+    print()
+
+
+def add_seed(seeds, limit=None):
+    unique_seeds = set(seeds)
+
+    running = True
+    while running:
+        # Find all song numbers matching a keyword
+        song_numbers = find_seed_song_numbers()
+        if not song_numbers:
+            break
+
+        # Select a song from the search results
+        song_number = select_seed_song_number(song_numbers)
+        if song_number is not None:
+            unique_seeds.add(song_number)
+            song = find_spotify_info(song_number)
+            print(f'Added: {song["name"]}')
+
+        if limit is not None and len(unique_seeds) >= limit:
+            print('Reached seed limit.')
+            running = False
+        else:
+            running = input_bool('Add another song?')
+
+    return sorted(unique_seeds)
+
+
+def remove_seed(seeds):
+    updated_seeds = seeds
+    running = True
+
+    while running:
+        print('Select the index of the song that you wish to remove: ')
+        (index, valid_index) = select_option(seeds, option_label_fn=get_song_option_label)
+
+        if not valid_index:
+            print(f'Sorry. There is no song at index "{index}".')
+            running = input_bool("Try another index?")
+
+        else:
+            song_number = updated_seeds.pop(index - 1)
+            song = find_spotify_info(song_number)
+            print(f'Removed: {song["name"]}')
+
+            if not updated_seeds:
+                print('No more songs to remove.')
+                running = False
+            else:
+                running = input_bool('Remove another song?')
+
+    return updated_seeds
+
+
 def workflow_2(playlist_length):
-    # TODO: Implementation of WF#2
-    return []
+    seeds = []
+    playlist_songs = []
+    finished_workflow = False
+    seed_limit = playlist_length
+
+    while not finished_workflow:
+        print('Please pick from the options below:')
+        print('[1] Display seeds')
+        if len(seeds) < seed_limit:
+            print('[2] Add song')
+        if seeds:
+            print('[3] Remove song')
+            print('[4] Generate playlist')
+        print('[0] Cancel')
+
+        option = input()
+
+        if option == '0':
+            return None
+
+        if option == '1':
+            display_seeds(seeds)
+            continue
+
+        if option == '2':
+            if len(seeds) < seed_limit:
```

Comment:
Included seed limit here. If the its above the seed limit this option cannot be selected